### ISSUE TITLE
Tensor shape difference when using attention mask.

### ISSUE THREAD

**Issue by aravind-h-v** [cubiq/ComfyUI_IPAdapter_plus#255]
This is a continuation of the issue

https://github.com/cubiq/ComfyUI_IPAdapter_plus/issues/130

which was marked closed. I noticed it is possible to trigger the error even if all the images are of the same shape and also divisible by 8. I have included the data to reproduce this in:

https://drive.google.com/file/d/1eJspLwGPuFr3ADvEKPc8S8HhU-bOEXqR/view?usp=sharing

The main problem seems to be in the portion

`               mask_h = lh / math.sqrt(lh * lw / qs)
`
`
                mask_h = int(mask_h) + int((qs % int(mask_h)) != 0)
`
`
                mask_w = qs // mask_h
`

I see that mask_h * mask_w has to equal qs and both have to be integers. I donot understand the full details of how the number qs comes or what are the constrains but if we have situations where such factorization is not possible, we run into this error. For instance, when

`lh =  128`
`lw =  45`
`qs =  384`

The decomposition here leads to 

`mask_h =  34`
`mask_w =  11`

but a better decomposition would have been

`mask_h =  32`
`mask_w =  12`

but either way, I can imagine circumstances where such decomposition might not be possible. So instead, can we add an interpolation function just before the line

`out_ip = out_ip * mask_downsample`

as:

` mask_downsample = F.interpolate(mask_downsample.unsqueeze(1), size=(out_ip.shape[1], out_ip.shape[2]), mode="bicubic").squeeze(1) `

This should solve the shape mismatch issue. But is this logic correct ?

**Comment by cubiq** [comment]
I'll work on it, I haven't forgotten.

**Comment by aravind-h-v** [comment]
Ya, thanks! Just wanted to share what I found!

**Comment by cubiq** [comment]
this should be fixed
